pkg/option/resolver/resolver.go

```diff
@@ -230,12 +231,54 @@ func readConfigMap(ctx context.Context, client client.Clientset, source ConfigSo
 	return cm.Data, []string{source.String()}, nil
 }
 
-// ReadConfigOverrides reads all the CiliumNodeConfig objects and returns a flattened map
+// readNodeConfigsAllVersions read node configurations for versions v2 and v2alpha1 of CiliumNodeConfig CRD.
+// TODO remove me when CiliumNodeConfig v2alpha1 is depreacted
+func readNodeConfigsAllVersions(ctx context.Context, client client.Clientset, nodeName, namespace, name string) (map[string]string, []string, error) {
+	var errv2, errv2alpha1 error
+	nodeConfigv2, descv2, errv2 := readNodeConfigs(ctx, client, nodeName, namespace, name)
+	if errv2 != nil {
+		log.WithFields(logrus.Fields{logfields.Node: nodeName}).Errorf("CiliumNodeConfig v2 not found: %s", errv2)
+	}
+
+	nodeConfigv2alpha1, descv2alpha1, errv2alpha1 := readNodeConfigsv2alpha1(ctx, client, nodeName, namespace, name)
+	if errv2alpha1 != nil {
+		log.WithFields(logrus.Fields{logfields.Node: nodeName}).Errorf("CiliumNodeConfig v2alpha1 not found: %s", errv2alpha1)
+
+		// Return the two errors
+		if errv2 != nil && errv2alpha1 != nil {
+			err := fmt.Errorf("CiliumNodeConfig v2 and v2alpha1 not found: %s and %s", errv2, errv2alpha1)
+			return nil, nil, err
+		}
+
+		return nil, nil, errv2alpha1
+	}
+
+	// Merge the two maps and keep v2 value in case of similar values
+	nodeConfig := make(map[string]string, 0)
```

Comment:
I tried to use `maps.Copy` but I need to check that there are values in both maps before copying otherwise I face a "panic: assignment to entry in nil map"... I wrote the following 
```
	// Merge the two maps and keep v2 value in case of similar keys
	nodeConfig := make(map[string]string, 0)
	if nodeConfigv2alpha1 != nil {
		maps.Copy(nodeConfig, nodeConfigv2alpha1)
	}
	if nodeConfigv2 != nil {
		maps.Copy(nodeConfig, nodeConfigv2)
	}
```

What do you think?

Reply:
That seems a bit strange.  If m1 is allocated and m2 is nil it should not bomb since you can always read from a nil map.

Reply:
I faced the following Go issue: https://github.com/golang/go/issues/64390
When the destination map is nil and the source isn't, the compiler returns a panic. 

I updated my code with the following snippet but I find it pretty ugly, wdyt this time?
```go
	if nodeConfigv2alpha1 == nil {
		nodeConfigv2alpha1 = nodeConfigv2
	} else {
		maps.Copy(nodeConfigv2alpha1, nodeConfigv2)
	}
```